src/views/catalog/customize/utils.ts

```diff
@@ -109,21 +109,22 @@ export const processTemplate = async ({
       virtualMachineName,
     );
 
-    if (withWindowsDrivers) {
-      draftTemplate = await mountWinDriversToTemplate(draftTemplate);
-    }
+    const draftVM = getTemplateVirtualMachineObject(draftTemplate);
+
+    draftVM.metadata.name = virtualMachineName;
   });
 
-  const processedTemplate = await k8sCreate<V1Template>({
+  if (withWindowsDrivers) {
+    templateToProcess = await mountWinDriversToTemplate(templateToProcess);
```

Comment:
Now `mountWinDriversToTemplate` already create another template copy. Inside the `produce` function was not working anymore.